lessons/basics/comprehensions.md

```diff
@@ -0,0 +1,107 @@
+---
+layout: page
+title: Comprehensions
+category: basics
+order: 13
+lang: en
+---
+
+List comprehensions are syntactic sugar for looping through enumerables in Elixir.  In this lesson we'll look at how we can use comprehensions for iteration and generation.
+
+## Table of Contents
+
+- [Basics](#basics)
+- [Filters](#filters)
+- [Using `:into`](#using-into)
+
+## Basics
+
+Often times comprehensions can be used to produce more concise statements for `Enum` and `Stream` iteration.  Let's start by looking at a simple comprehension and then break it down:
+
+```elixir
+iex> list = [1, 2, 3, 4, 5]
+iex> for x <- list, do: x*x
+[1, 4, 9, 16, 25]
+```
+
+The first thing we notice is the use of `for` and a generator.  What is a generator?  Generators are the `x <- [1, 2, 3, 4]` expressions found in list comprehensions, they're responsible for generating the next value.
+
+Lucky for us comprehensions aren't limited to lists, in fact they'll work with any enumerable:
+
```

Comment:
I think would be nice a little explanation that `comprehensions` can work with pattern matching, it's clear in the code but as it's a basic lesson and the order is 13, would be good to explain that as well.